Sources/BSON/Document.swift

```diff
@@ -1,6 +1,33 @@
 import Foundation
 import NIO
 
+/// This shared allocator instance should be used for all underlying `ByteBuffer` creation.
+private let BSON_ALLOCATOR = ByteBufferAllocator()
+
+extension ByteBuffer {
+    @discardableResult
+    internal mutating func writeCString(_ string: String) -> Int {
+        let written = self.writeString(string + "\0")
+        return written
+    }
+
+    internal mutating func readCString() throws -> String? {
+        var string = ""
+        for i in 0..<0xFFFFFE {
```

Comment:
is this corresponding to max BSON document size? consider introducing a constant for that we can reference throughout the codebase